### ISSUE TITLE
Definitions created using 'az synapse spark-job-definition create' do not show in the Workspace UI

### ISSUE THREAD

**Issue by alexszym** [Azure/azure-cli#23348]
**Related command**
`az synapse spark-job-definition create --workspace-name <workspace> --name <job-name> --file @"folder\sparkjob.json"`

**Describe the bug**
When using 'az synapse spark-job-definition create' command I am able to create a sparkjob definition, however that definition is not then visible in the Synapse Workspace UI. I am however able to see it when calling 'az synapse spark-job-definition list'

I have Git integration enabled on that workspace, checked all the branches and can't find the job definition I've created in the UI.

**To Reproduce**
1. Export existing job definition from Synapse UI
2. Change jobProperties.name to be unique.
3. Run `az synapse spark-job-definition create --workspace-name <workspace> --name <job-name> --file @"folder\sparkjob.json" using the previously exported file.
4. Run `az synapse spark-job-definition list --workspace-name <workspace>` and verify newly create job definition is visible.
5. Check UI.

**Expected behavior**
The created spark job definition is also visible in Synapse Workspace UI

**Environment summary**
```
Windows-10-10.0.19044-SP0
Python 3.10.5
Installer: MSI

azure-cli 2.38.0

Extensions:
azure-cli-ml 1.41.0

Dependencies:
msal 1.18.0b1
azure-mgmt-resource 21.1.0b1
```

**Additional context**

**Comment by RakeshMohanMSFT** [comment]
@alexszym Thank you for reaching out, we are looking into it.

**Comment by yonzhan** [comment]
route to CXP team

**Comment by RakeshMohanMSFT** [comment]
@alexszym We are unable to reproduce the issue. Here is the command and sample file we used.

Ensure that the folder path and pool reference matches

`az synapse spark-job-definition create --workspace-name issue23348workspace --name issue23348jobdefinition --file @test.json --folder-path "issue23348fs/Jobs"`

{
  "description": null,
  "folder": {
    "name": "issue23348fs/Jobs"
  },
  "jobProperties": {
    "args": [
        "-i",
        "tst",
        "-n",
        "Enrichment1"
    ],
    "file": "abfss://jobdeveloper@{adlsAccount}.dfs.core.windows.net/issue23348main.py",
    "files": [
      "abfss://jobdeveloper@{adlsAccount}.dfs.core.windows.net/issue23348.py"
    ],
    "numExecutors": 2,
    "driverCores": 4,
    "driverMemory": "28g",
    "executorCores": 4,
    "executorMemory": "28g"
  },
  "language": "python",
  "requiredSparkVersion": "3.2",
  "targetBigDataPool": {
    "referenceName": "issue2334sparkp",
    "type": "BigDataPoolReference"
  }
}

![image](https://user-images.githubusercontent.com/49954584/181470532-64790060-e4a7-41f0-b19b-c78fdd301c1f.png)

**Comment by alexszym** [comment]
Hi @RakeshMohanMSFT,

I've tried with the sample file and I'm still seeing the same issue on my side. Can you confirm for me if you've tested with Git integration enabled on the workspace and which branch does the job definition appear on for you?

**Comment by RakeshMohanMSFT** [comment]
@alexszym Apologies, i missed the git part, let me try again and get back to you

**Comment by alexszym** [comment]
@RakeshMohanMSFT I've tried the same script on my side without Git integration enabled and that works fine, so I think the cause of the issue might indeed be the Git integration feature.

In case its relevant, my branches are named:
- `main` (main branch)
- `publish` (publish branch)

**Comment by RakeshMohanMSFT** [comment]
@alexszym I have made the git configuration similar to yours. Please check once and also if possible give me a sample template of the file you are using for job definition

![image](https://user-images.githubusercontent.com/49954584/181597714-fd589249-192d-432e-baf1-6292913d1d69.png)

**Comment by alexszym** [comment]
@RakeshMohanMSFT In my Git configuration I'm using Azure DevOps rather than GitHub in case that makes a difference. The file I've used is very similar to yours. It worked fine on the Synapse workspace without Git integration.

```
{
  "description": null,
  "folder": {
    "name": "issue23348fs/Jobs"
  },
  "jobProperties": {
    "args": [
      "-i",
      "tst",
      "-n",
      "Enrichment1"
    ],
    "file": "abfss://jobdeveloper@{adlsAccount}.dfs.core.windows.net/issue23348main.py",
    "files": [
      "abfss://jobdeveloper@{adlsAccount}.dfs.core.windows.net/issue23348.py"
    ],
    "numExecutors": 2,
    "driverCores": 4,
    "driverMemory": "28g",
    "executorCores": 4,
    "executorMemory": "28g"
  },
  "language": "python",
  "requiredSparkVersion": "3.2",
  "targetBigDataPool": {
    "referenceName": "testpool",
    "type": "BigDataPoolReference"
  }
}
```
Command:
`az synapse spark-job-definition create --workspace-name <workspace> --name TestSparkjob --file @Test.json --folder-path "issue23348fs/Jobs"`

**Comment by RakeshMohanMSFT** [comment]
@alexszym The job definitions created using command line get created in our synapse live studio and there does not seem to be a way available to push them to github directly. But they do become visible in the synapse live. 

But if you disconnect and reconnect the github and import again they are getting merged in the github.

![image](https://user-images.githubusercontent.com/49954584/181699319-e1a04c1a-01b5-4be1-9886-a8f8b0ad981a.png)

**Comment by alexszym** [comment]
Hi @RakeshMohanMSFT, thanks for getting back to me. I can indeed see the jobs I've created when I swap to Synapse Live mode. 

Any idea how much of the synapse CLI might be affected by this issue? I've been able to use `spark pool` and `workspace-package` commands successfully, without encountering it. If the rest of the CLI is likely to have the same issue (notebooks, pipelines etc), then having Git integration enabled means the CLI is not usable. Disconnecting and reconnecting the integration is not a viable solution in the long term. Might be worth updating the documentation?

Also, any idea which team I should raise the support ticket with? Is it the Synapse team itself?

**Comment by RakeshMohanMSFT** [comment]
@alexszym Let me engage the synapse engineering team (CLI), before we resort to support.

**Comment by alexszym** [comment]
Any updates on this issue?

**Comment by wonner** [comment]
> Any updates on this issue?

@alexszym Sorry, I missed this thread before. @kevinzz6 please help investigate. Thanks.

**Comment by kevinzz6** [comment]
Hi @alexszym, I tried to use cli to create other artifacts like SQL script with git enable and there is the same issue as you mentioned above.  But something like spark pools that are managed by "azure-mgmt-synapse" module are good. So, I think this issue is related to git CI/CD background.

**Comment by alexszym** [comment]
Thanks @kevinzz6, anything more on that issue?
